odoo/addons/base/models/ir_http.py

```diff
@@ -278,7 +278,11 @@ def routing_map(cls, key=None):
                 kw = {k: routing[k] for k in xtra_keys if k in routing}
                 rule = werkzeug.routing.Rule(url, endpoint=endpoint, methods=routing['methods'], **kw)
                 rule.merge_slashes = False
-                routing_map.add(rule)
+                try:
```

Comment:
This should probably not be here in regular dispatching process.
We still want it to crash if we introduce a incorrect controller in business code IMHO.

I'd move it to the `_check_url_to` constraint (and I think that's what JKE meant by `try to load the new rule in a copy of current routing table`.
Something like
```python
try:
    rule = werkzeug.routing.Rule(rewrite.url_to)
    werkzeug.routing.Map(strict_slashes=False).add(rule)
except ValueError as e:
    raise ValidationError(_('"%s" is not a valid URL: %s>' % (rewrite.url_to, e)))
```

Reply:
What you suggest is very close to what I did initially (see https://github.com/odoo/odoo/commit/43d390133210d58b123331c1fd2084b679cccc4a)
But I got the following feedback from @JKE-be:
- it was too violent, therefore I just turned it to a basic validation
- that we should make sure a wrong configuration does not block the system from being used at all (which was the initial issue), therefore I neutralized the exception in the routing map building, so that other routes would still be usable

I could restore such a validation, but I fear it will still be rejected.